src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -57,14 +60,26 @@ protected RobotState getNextState(RobotState currentState) {
               IDLE_NO_GP,
               IDLE_WITH_GP,
               CLIMBING_1_LINEUP,
-              CLIMBING_2_HANGING ->
+              CLIMBING_2_HANGING,
+              PODIUM_WAITING,
+              OUTTAKING ->
           currentState;
-      case SPEAKER_SCORING, FEEDING_SHOOTING, PASS_SHOOTING, SUBWOOFER_SCORING ->
+      case SPEAKER_SCORING,
+              AMP_SCORING,
+              FEEDING_SHOOTING,
+              PASS_SHOOTING,
+              SUBWOOFER_SCORING,
+              PODIUM_SCORING ->
           queuer.hasNote() ? currentState : RobotState.IDLE_NO_GP;
-      case AMP_SCORING -> queuer.hasNote() ? currentState : RobotState.AMP_WAITING;
 
       case SPEAKER_PREPARE_TO_SCORE ->
-          shooter.atGoal() && arm.atGoal() ? RobotState.SPEAKER_SCORING : currentState;
+          (shooter.atGoal() && arm.atGoal() && DriverStation.isTeleop() == false)
+                  || (DriverStation.isTeleop()
+                      && confirmShotActive == true
```

Comment:
you can write this logic but it's probably easier to just say `is autonomous OR confirm shot active`